cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/ConnectorStatusUpdater.java

```diff
@@ -43,4 +81,23 @@ public void update(ManagedConnector connector) {
         }
     }
 
+    /* Expose a metric "cos_fleetshard_sync_connector_state" which reveals the current connector state.
+    Metric value of 1 implies that the connector is in Ready state. Similarly, 2 -> Failed, 3 -> Deleted,
+    4 -> Stopped, 5 -> In Process */
+    private void measure(ManagedConnector connector, Integer connectorState) {
+
+        tags = List.of(
```

Comment:
you must not use an instance variable here as if this method get invoked my multiple threads, then you may get unpredictable results